lib/s3routes/routesUtils.ts

```diff
@@ -163,10 +162,11 @@ export const XMLResponseBackend = {
         const xmlStr = xml.join('');
         const bytesSent = Buffer.byteLength(xmlStr);
         log.addDefaultFields({ bytesSent });
-        response.writeHead(error.code, {
+        response.removeAllListeners('finish');
```

Comment:
similar to setting headers & result codes, can't we always set the finish handler (or `end()` callback) in a single place, when sending the response, instead of "pre-setting" it and somehow replacing it when it is not appropriate?